www/js/model.js

```diff
@@ -981,6 +993,19 @@ define([
 					if(genus == "requred" || genus == "allowed")
 						arr.push(node);
 				});
+			},
+			getDescriptionsSortedByName: function(){
+				var descriptions = obj.active.getDescriptions();
+				var descNameMap = array.map(descriptions, function (desc) {
+					var authoredID = obj.student.getAuthoredID(desc.value);
+					var _name = obj.authored.getName(authoredID);
+					return {name: _name, description: desc.label, id: desc.value};
+				}, this);
+				descNameMap.sort(function (obj1, obj2) {
```

Comment:
These two functions can be merged with a simple check. I don't think there is a need for two separate functions.